pytorch_lightning/loops/batch/manual.py

```diff
@@ -72,12 +73,25 @@ def advance(self, batch: Any, batch_idx: int) -> None:  # type: ignore[override]
 
             _check_training_step_output(ligtning_module, training_step_output)
 
-            result_collection, self._hiddens = _process_training_step_output(self.trainer, training_step_output)
+            self._hiddens = _extract_hiddens(training_step_output)
+
+            # TODO: do not use `ClosureResult`
```

Comment:
What the plan instead ?

Reply:
See discussion here: https://github.com/PyTorchLightning/pytorch-lightning/pull/9349#discussion_r704843554